src/device/wg_controller.rs

```diff
@@ -905,6 +911,7 @@ fn peer_state(
     time_since_last_rx: Option<&Duration>,
     time_since_last_endpoint_change: Option<&Duration>,
     proxy_endpoint: Option<&SocketAddr>,
+    replaced_proxy_endpoint: Option<&SocketAddr>,
```

Comment:
I wouldn't be passing `SocketAddr` by reference but if you want to be consistent with the code around, then it's somewhat fine

Reply:
I passed it by reference to be consistent, but I also tried just now to just deref it and my terminal overflowed with trait bound issues, so I will be keeping it as a ref